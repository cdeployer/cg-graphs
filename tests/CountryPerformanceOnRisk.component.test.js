import { CountryPerformanceOnRisk, CountrySelect }
from '../src/components/CountryPerformanceOnRisk.js';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';


describe('Components are working fine', () => {

  it('computeState works', () => {
    const wrapper = shallow(< CountryPerformanceOnRisk countries={['Is Working']}/>)
    let out = wrapper.instance().computeState()
    expect(out.countries).toEqual(['Is Working'])
  })

  it('Div with id DDOS-graph is there', () => {
    const wrapper = shallow(< CountryPerformanceOnRisk />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  // it('When a country is selected, it updates state of the container', () => {
  //   function stub() {return ''}
  //   const wrapper = shallow(< CountryPerformanceOnRisk dispatch={stub} />)
  //   let newCountry = {value: 'us', label: 'United States'}
  //   expect(wrapper.find('CountrySelect').at(2).props().selectedCountry)
  //     .toEqual(undefined)
  //   wrapper.find('CountrySelect').at(2).simulate('change', newCountry)
  //   expect(wrapper.find('CountrySelect').at(2).props().selectedCountry)
  //     .toEqual(newCountry)
  // })

})

describe('Country Select component', () => {
  function setup() {
    let props = {
      selectOptions: [
        {value: '', label: 'Select a country'},
        {value: 'uk', label: 'United Kingdom' }
      ],
      onChange: jest.fn(),
      selectedCountry: undefined
    }

    const enzymeWrapper = shallow(< CountrySelect {...props} />)

    return {
      props,
      enzymeWrapper
    }
  }

  it('Dropdown is there', () => {
    const { enzymeWrapper } = setup()
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  })

  it('Selecting a country triggers event on parent component', () => {
    const { enzymeWrapper, props } = setup()
    const selector = enzymeWrapper.find('Select')
    expect(selector.props().value).toEqual({
      "label": "Select a country", "value": ""
    })
    let selectedOption = {value: 'uk', label: 'United Kingdom' }
    selector.props().onChange(selectedOption)
    expect(props.onChange.mock.calls.length).toEqual(1)
    expect(props.onChange.mock.calls[0][0]).toEqual(selectedOption)
  })
})
