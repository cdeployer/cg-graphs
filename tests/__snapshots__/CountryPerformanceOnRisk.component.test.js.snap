exports[`Components are working fine Renders expected DOM 1`] = `
<div
  className="graph-div">
  <div
    className="row">
    <div
      className="col-sm-11">
      <h3>
        OPEN DNS
           |  
        UNITED KINGDOM
      </h3>
    </div>
    <div
      className="col-sm-1" />
  </div>
  <PlotlyGraph
    data={Array []}
    graphOptions={
      Object {
        "font": Object {
          "color": "#7f7f7f",
          "size": 9,
        },
        "height": 200,
        "legend": Object {
          "x": 0,
          "y": 1,
        },
        "margin": Object {
          "b": 30,
          "l": 40,
          "r": 30,
          "t": 0,
        },
        "xaxis": Object {
          "gridcolor": "transparent",
        },
        "yaxis": Object {
          "title": undefined,
        },
      }
    } />
  <form
    className="radio-form">
    <label
      className="radio-inline">
      <input
        checked={true}
        onChange={[Function]}
        type="radio"
        value="count_normalized" />
      Simple counts
    </label>
    <label
      className="radio-inline">
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="count_normalized" />
      Trend
    </label>
  </form>
</div>
`;

exports[`Country Select component Dropdown is there 1`] = `
<div
  className="Select-div">
  <Select
    addLabelText="Add \"{label}\"?"
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    name="countries"
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onInputChange={[Function]}
    openAfterFocus={false}
    optionComponent={[Function]}
    optionRenderer={[Function]}
    options={
      Array [
        Object {
          "label": "Select a country",
          "value": "",
        },
        Object {
          "label": "Global",
          "value": "t",
        },
        Object {
          "label": "Georgia",
          "value": "ge",
        },
      ]
    }
    pageSize={5}
    placeholder="Select..."
    required={false}
    scrollMenuIntoView={true}
    searchable={true}
    simpleValue={false}
    tabSelectsValue={true}
    value={
      Object {
        "label": "Select a country",
        "value": "",
      }
    }
    valueComponent={[Function]}
    valueKey="value" />
</div>
`;
